src/pubsub.ts

```diff
@@ -48,6 +48,8 @@ export class AMQPPubSub implements PubSubEngine {
 
   public async subscribe(routingKey: string, onMessage: (message: any) => void): Promise<number> {
     const id = this.currentSubscriptionId++;
+    logger('Subscribe id: "%s"', id);
```

Comment:
What's the reasoning behind this log line?

Reply:
While triaging issues in this repo, I had a hard time debugging sub/unsub issues without knowing what `id` was being used for it. After adding this in and the unsub equivalent, I was better able to track down what top level API calls in a specific pattern would cause race conditions.

I can remove here and in unsub if you think it adds too much noise.

Reply:
If we could change the message to something that would be useful to someone activating the logs I would gladly leave this in.
I'm thinking about something like this:
`logger('Subscribing to "%s" with id: "%s"', routingKey, id);`
Line is untested.

So if everything is working as expected you get messages in the log telling you about every new subscribe.
Helping you to notice errors in your application and not just the library.
And if it needed to open a new subscription to the rabbitmq it will be followed up by a message like this.
`2020-05-12T12:35:43.899Z AMQPPubSub Subscribed to Queue "amq.gen-1aGd11coG0Kt_6QTvcTdfA" (amq.ctag-mnstYib1qtZyudeWKYwPcA)`

Same thing for unsubscribe.

Reply:
Switched to new message format.